src/pages/material/material-details/material-details.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from 'react';
+import { withRouter } from 'react-router';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { useStyles } from '../material-add/material-add.styles';
+import LoadingBar from '../../../components/loading-bar';
+import MaterialForm from '../../../components/material-form';
+import { getMaterial } from '../../../redux/material/material.actions';
+
+const MaterialDetails = ({ match }) => {
+  const { id } = match.params;
+
+  const dispatch = useDispatch();
+
+  const { loading, material } = useSelector(({ Material }) => ({
+    loading: Material.materialLoading,
+    material: Material.material
+  }));
+  const styles = useStyles();
+
+  useEffect(() => {
+    dispatch(getMaterial(id));
+  }, [dispatch, id]);
+
+  if (loading) {
+    return <LoadingBar />;
+  }
+  return (
+    <div className={styles.container}>
+      {material !== null ? <MaterialForm id={id} material={material} /> : null}
+    </div>
+  );
+};
+
+const valueShape = PropTypes.shape({
+  value: PropTypes.string
+});
+
+MaterialDetails.propTypes = {
+  match: PropTypes.shape({
+    params: PropTypes.shape({
+      id: PropTypes.string.isRequired
+    })
+  }).isRequired,
+  material: PropTypes.shape({
+    _id: PropTypes.string,
+    name: PropTypes.arrayOf(valueShape),
+    description: PropTypes.arrayOf(valueShape),
+    images: PropTypes.shape({
+      thumbnail: PropTypes.string
+    }),
+    purpose: PropTypes.string,
+    available: PropTypes.bool
+  })
+};
+
+MaterialDetails.defaultProps = {
+  material: {}
+};
+
+export default withRouter(MaterialDetails);
```

Comment:
You don't need this HOC, cause we use `connected-react-router`, that are connected to redux and browser history

Reply:
Fixed